src/modm/math/geometry/location_2d.hpp

```diff
@@ -37,39 +36,60 @@ namespace modm
 	class Location2D
 	{
 	public:
-		Location2D();
-
-		Location2D(const Vector<T, 2>& position, const float& orientation);
-
-		Location2D(const T& x, const T& y, const float& orientation);
-
-		inline const Vector<T, 2>&
-		getPosition() const;
-
-		inline const T&
-		getX() const;
-
-		inline const T&
-		getY() const;
-
-		void
-		setPosition(const Vector<T, 2>& point);
-
-		void
-		setPosition(const T& x, const T& y);
+		Vector<T, 2> position;
+		float orientation = 0;
+
+		constexpr Location2D() = default;
+		constexpr Location2D(const Vector<T, 2>& position, const float& orientation)
+			: position(position), orientation(orientation) {}
+
+		[[deprecated("Use 'setPosition({x, y}, orientation)' instead!")]]
+		constexpr Location2D(const T& x, const T& y, const float& orientation)
+			: position(x, y), orientation(orientation) {}
+
+		template<typename U>
+		constexpr Location2D(const Location2D<U> &l) : position(l.position), orientation(l.orientation) {}
```

Comment:
This enables silent and implicit narrowing conversions. This constructor should be explicit.